index.js

```diff
@@ -46,12 +60,16 @@ module.exports = function (str, opts) {
 	}
 
 	// remove duplicate slashes
-	urlObj.pathname = urlObj.pathname.replace(/\/{2,}/, '/');
+	if (urlObj.pathname) {
+		urlObj.pathname = urlObj.pathname.replace(/\/{2,}/, '/');
+	}
 
-	// resolve relative paths
-	var domain = urlObj.protocol + '//' + urlObj.hostname;
-	var relative = url.resolve(domain, urlObj.pathname);
-	urlObj.pathname = relative.replace(domain, '');
+	// resolve relative paths, but only for slashed protocols
+	if (slashedProtocol[urlObj.protocol]) {
```

Comment:
I tried, but I noticed that prependHttp('mailto:sindre@sorhus.com?subject=ponny+er+best') fails and adds http in front. Should it really prepend http on valid urls that already contains a scheme?